### ISSUE TITLE
[Bug] dgl.to_homo fails on given example in pytorch 1.3.1

### ISSUE THREAD

**Issue by modesitt-sesco** [dmlc/dgl#1018]
## To Reproduce

Simply run the `to_homo` example 

```
import dgl 

follows_g = dgl.graph([(0, 1), (1, 2)], 'user', 'follows')
devs_g = dgl.bipartite([(0, 0), (1, 1)], 'developer', 'develops', 'game')
hetero_g = dgl.hetero_from_relations([follows_g, devs_g])
homo_g = dgl.to_homo(hetero_g)
```

and it fails with 

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/dmodesitt/Dev/playground/virt/venv/lib/python3.7/site-packages/dgl/convert.py", line 598, in to_homo
    retg = graph((F.cat(srcs, 0), F.cat(dsts, 0)), card=total_num_nodes)
  File "/Users/dmodesitt/Dev/playground/virt/venv/lib/python3.7/site-packages/dgl/convert.py", line 111, in graph
    return create_from_edges(u, v, ntype, etype, ntype, urange, vrange)
  File "/Users/dmodesitt/Dev/playground/virt/venv/lib/python3.7/site-packages/dgl/convert.py", line 646, in create_from_edges
    u = utils.toindex(u)
  File "/Users/dmodesitt/Dev/playground/virt/venv/lib/python3.7/site-packages/dgl/utils.py", line 242, in toindex
    return data if isinstance(data, Index) else Index(data)
  File "/Users/dmodesitt/Dev/playground/virt/venv/lib/python3.7/site-packages/dgl/utils.py", line 15, in __init__
    self._initialize_data(data)
  File "/Users/dmodesitt/Dev/playground/virt/venv/lib/python3.7/site-packages/dgl/utils.py", line 22, in _initialize_data
    self._dispatch(data)
  File "/Users/dmodesitt/Dev/playground/virt/venv/lib/python3.7/site-packages/dgl/utils.py", line 47, in _dispatch
    raise DGLError('Index data must be an int64 vector, but got: %s' % str(data))
dgl._ffi.base.DGLError: Index data must be an int64 vector, but got: tensor([0., 1., 3., 4.])
```

## Expected Behavior 

This is not expected to crash it is a given [example](https://docs.dgl.ai/generated/dgl.to_homo.html#dgl.to_homo)

## Environment

 - DGL Version (e.g., 1.0):
      dgl==0.4.1
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3):
      pytorch==1.3.1
 - OS (e.g., Linux):
     macos
 - How you installed DGL (`conda`, `pip`, source):
     pip
 - Python version: 3.7.4

**Comment by mufeili** [comment]
Verified the report. This seems to be related to the latest PyTorch version as our CI tests with PyTorch 1.1 can be successfully passed.

**Comment by mufeili** [comment]
This should now be fixed in #1019 . You can install from source to get the latest version.

**Comment by lzzhaha** [comment]
> This should now be fixed in #1019 . You can install from source to get the latest version.

Hi, 

Thanks for your efforts. But I don't have root priviledge for the servers I am working on. So I cannot use 'sudo' to install from source...

Could you please provide me with any advice for getting around with this issue?

Thank you very much!

**Comment by mufeili** [comment]
@lzzhaha Can you install new packages with pip or Conda? If so, you can install DGL v0.4.2 to address the issue.

**Comment by lzzhaha** [comment]
@mufeili Yes the problem is indeed solved in DGL v0.4.2 with pip installation. 
Thank you so much!
